# Import all required library of python

In [1]:
import numpy as np 
import pandas as pd

# Download Titanic dataset from kaggle

In [2]:
data=pd.read_csv("tested.csv")

# Data Preprocessing

info shows datatype and null values of each column

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    object 
 4   Sex          418 non-null    object 
 5   Age          332 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    object 
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     object 
 11  Embarked     418 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 39.3+ KB


Describe functin is used for calculating some statistical data

In [4]:
data.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   418.000000  418.000000  418.000000  332.000000  418.000000   
mean   1100.500000    0.363636    2.265550   30.272590    0.447368   
std     120.810458    0.481622    0.841838   14.181209    0.896760   
min     892.000000    0.000000    1.000000    0.170000    0.000000   
25%     996.250000    0.000000    1.000000   21.000000    0.000000   
50%    1100.500000    0.000000    3.000000   27.000000    0.000000   
75%    1204.750000    1.000000    3.000000   39.000000    1.000000   
max    1309.000000    1.000000    3.000000   76.000000    8.000000   

            Parch        Fare  
count  418.000000  417.000000  
mean     0.392344   35.627188  
std      0.981429   55.907576  
min      0.000000    0.000000  
25%      0.000000    7.895800  
50%      0.000000   14.454200  
75%      0.000000   31.500000  
max      9.000000  512.329200

head funtion shows first 5 rows of data

In [5]:
data.head()

PassengerId  Survived  Pclass  \
0          892         0       3   
1          893         1       3   
2          894         0       2   
3          895         0       3   
4          896         1       3   

                                           Name     Sex   Age  SibSp  Parch  \
0                              Kelly, Mr. James    male  34.5      0      0   
1              Wilkes, Mrs. James (Ellen Needs)  female  47.0      1      0   
2                     Myles, Mr. Thomas Francis    male  62.0      0      0   
3                              Wirz, Mr. Albert    male  27.0      0      0   
4  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female  22.0      1      1   

    Ticket     Fare Cabin Embarked  
0   330911   7.8292   NaN        Q  
1   363272   7.0000   NaN        S  
2   240276   9.6875   NaN        Q  
3   315154   8.6625   NaN        S  
4  3101298  12.2875   NaN        S

shape function tells us about how many columns and rows are in data

In [6]:
data.shape

(418, 12)

It counts total number of null value in each column

In [7]:
data.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age             86
SibSp            0
Parch            0
Ticket           0
Fare             1
Cabin          327
Embarked         0
dtype: int64

So there are 86 null value in Age and 327 null value in Cabin column 

We fill all null value of Age column with median of column and fare with mean of column

In [8]:
data['Age'].fillna(data['Age'].median(),inplace=True)

In [9]:
data['Fare'].fillna(data['Fare'].mean(),inplace=True)

In [10]:
data.isnull().any()

PassengerId    False
Survived       False
Pclass         False
Name           False
Sex            False
Age            False
SibSp          False
Parch          False
Ticket         False
Fare           False
Cabin           True
Embarked       False
dtype: bool

As we know column Cabin is object(string) datatype so we can't perform any statistical approach

# Now we are going to seprate dependent and independent variables

In [11]:
data.head()

PassengerId  Survived  Pclass  \
0          892         0       3   
1          893         1       3   
2          894         0       2   
3          895         0       3   
4          896         1       3   

                                           Name     Sex   Age  SibSp  Parch  \
0                              Kelly, Mr. James    male  34.5      0      0   
1              Wilkes, Mrs. James (Ellen Needs)  female  47.0      1      0   
2                     Myles, Mr. Thomas Francis    male  62.0      0      0   
3                              Wirz, Mr. Albert    male  27.0      0      0   
4  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female  22.0      1      1   

    Ticket     Fare Cabin Embarked  
0   330911   7.8292   NaN        Q  
1   363272   7.0000   NaN        S  
2   240276   9.6875   NaN        Q  
3   315154   8.6625   NaN        S  
4  3101298  12.2875   NaN        S

In [12]:
x = data.iloc[:,:3]
x

PassengerId  Survived  Pclass
0            892         0       3
1            893         1       3
2            894         0       2
3            895         0       3
4            896         1       3
..           ...       ...     ...
413         1305         0       3
414         1306         1       1
415         1307         0       3
416         1308         0       3
417         1309         0       3

[418 rows x 3 columns]

In [13]:
y=data.iloc[:,8:10]
y

Ticket      Fare
0                330911    7.8292
1                363272    7.0000
2                240276    9.6875
3                315154    8.6625
4               3101298   12.2875
..                  ...       ...
413           A.5. 3236    8.0500
414            PC 17758  108.9000
415  SOTON/O.Q. 3101262    7.2500
416              359309    8.0500
417                2668   22.3583

[418 rows x 2 columns]

value_counts() function returns object containing counts of unique values

In [14]:
data["Embarked"].value_counts()

S    270
C    102
Q     46
Name: Embarked, dtype: int64

In [15]:
data["Sex"].value_counts()

male      266
female    152
Name: Sex, dtype: int64

Counter counts that how many male and female are there in column "Sex"

In [16]:
from collections import Counter as c
c(data["Sex"])

Counter({'male': 266, 'female': 152})

In [17]:
data1 =data.copy()

# Encode the columns which are categorical

In [18]:
from sklearn.preprocessing import LabelEncoder

In [19]:
le=LabelEncoder()
print("Before label encoding",c(data["Embarked"]))
data['Embarked']=le.fit_transform(data['Embarked'])
print("After label encoding",c(data["Embarked"]))

Before label encoding Counter({'S': 270, 'C': 102, 'Q': 46})
After label encoding Counter({2: 270, 0: 102, 1: 46})


In [20]:
data1['Sex']=data1['Sex'].replace(['male','female'],['0','1'])
data1["Sex"]

0      0
1      1
2      0
3      0
4      1
      ..
413    0
414    1
415    0
416    0
417    0
Name: Sex, Length: 418, dtype: object

In [21]:
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder

In [22]:
ct=ColumnTransformer([("one",OneHotEncoder(),[0])],remainder="passthrough")
x

PassengerId  Survived  Pclass
0            892         0       3
1            893         1       3
2            894         0       2
3            895         0       3
4            896         1       3
..           ...       ...     ...
413         1305         0       3
414         1306         1       1
415         1307         0       3
416         1308         0       3
417         1309         0       3

[418 rows x 3 columns]

In [23]:
x.shape

(418, 3)

In [24]:
x=ct.fit_transform(x)
x

<418x420 sparse matrix of type '<class 'numpy.float64'>'
	with 988 stored elements in Compressed Sparse Row format>

# Splitting traing and testing data

In [26]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=0)

In [27]:
print(x_train.shape)
print(x_test.shape)
print(y_train.shape)
print(y_test.shape)

(292, 420)
(126, 420)
(292, 2)
(126, 2)
